### PR TITLE
fix: hide internal analytics commands from command palette

### PR DESCRIPTION
**Pull request by mathur07**
fixes https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension/issues/512

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -158,6 +158,20 @@
           "command": "extension.fabric8AnalyticsWidgetFullStackFromEditor",
           "when": "resourceFilename == go.mod"
         }
+      ],
```
Comment (arajkumar) [inline package.json:161]: Can you add "Dependency Analytic" category to the commands?
https://code.visualstudio.com/api/references/contribution-points#contributes.commands

```
"commands": [
      {
        "command": "fabric8.stackAnalysis",
        "title": "Full Stack Analysis...",
        "category": "Dependency Analytic"
      },
      {
        "command": "extension.fabric8AnalyticsWidgetFullStackFromPieBtn",
        "title": "Open Vulnerability Report",
        "icon": {
          "light": "icon/report-icon.png",
          "dark": "icon/report-icon.png"
        }
      },
```

Also I guess the following commands shall be removed, as it is registered programatically.

```
 {
        "command": "extension.fabric8AnalyticsWidgetFullStackFromExplorer",
        "title": "Dependency Analytics Report..."
      },
      {
        "command": "extension.fabric8AnalyticsStackLogs",
        "title": "Dependency Analytics Logs ..."
      }
```

Comment (mathur07) [reply package.json:161]: ack

Comment (mathur07) [reply package.json:163]: @arajkumar I have added the category, but have not removed the commands. those commands don't seem to be registered programmatically.

File: package.json [hunk 2]
```diff
@@ -77,7 +77,8 @@
     "commands": [
       {
         "command": "extension.fabric8AnalyticsWidgetFullStack",
-        "title": "Dependency Analytics Report ..."
+        "title": "Dependency Analytics Report...",
+        "category": "Dependency Analytics"
```
Comment (arajkumar) [inline package.json:81]: Now your command title shouldn't have "Dependency Analytics", it will be redundant.

File: package.json [hunk 3]
```diff
@@ -77,7 +77,8 @@
     "commands": [
       {
         "command": "extension.fabric8AnalyticsWidgetFullStack",
```
Comment (arajkumar) [inline package.json:79]: nit:

Please fix the command name as well.

e.g.
```
fabric8.stackAnalysis
```

File: package.json [hunk 4]
```diff
@@ -97,7 +98,8 @@
       },
       {
         "command": "extension.fabric8AnalyticsStackLogs",
-        "title": "Dependency Analytics Logs ..."
+        "title": "Dependency Analytics Logs...",
```
Comment (arajkumar) [inline package.json:101]:
```suggestion
        "title": "Debug logs",
```

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (mathur07) [reply package.json:81]: ack

File: package.json (hunk 3 above)
Comment (mathur07) [reply package.json:79]: ack

File: package.json (hunk 4 above)
Comment (mathur07) [reply package.json:101]: ack

File: package.json [hunk 5]
```diff
@@ -76,88 +76,104 @@
     ],
     "commands": [
       {
-        "command": "extension.fabric8AnalyticsWidgetFullStack",
-        "title": "Dependency Analytics Report ..."
+        "command": "extension.stackAnalysis",
```
Comment (arajkumar) [inline package.json:79]: Please fix the prefix as well, I had mentioned in the previous comment.

extension.stackAnalysis => fabric8.stackAnalysis or crda.stackAnalysis

File: test/resources/sampleNodeApp/package.json [hunk 6]
```diff
@@ -50,7 +50,7 @@
   },
```
Comment (arajkumar) [inline test/resources/sampleNodeApp/package.json:50]: These changes are not required, this is just a test project!

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (mathur07) [reply package.json:79]: ack

File: test/resources/sampleNodeApp/package.json (hunk 6 above)
Comment (mathur07) [reply test/resources/sampleNodeApp/package.json:50]: ack

**Review by arajkumar — APPROVED** [review]
lgtm

You can probably create a separate PR to update the respective images.

**Comment by mathur07** [comment]
ack

**Pull request merged**